### ISSUE TITLE
confusing understand loss

### ISSUE THREAD

**Issue by txdat** [experiencor/basic-yolo-keras#170]
hi, i read some others using iou_loss=no_obj*(0-iou_pred)*iou_pred + obj*(1-iou_pred)*overlap, different to yours. and in 1 batch (=12-16 images, due to maximum images for gpu's memory) very few anchors are assigned to be positive (equal number of objects in image per batch) so target of bounding boxes and classes are very sparse, contain lots of 0. is it problem for training network and for classification what is better loss (MSE or cross-entropy)? thanks!

**Comment by experiencor** [comment]
Can you point me to those other works? I closely follow the paper (https://pjreddie.com/media/files/papers/yolo_1.pdf) regarding this matter.

The sparsity of labels is a challenge and from the result so far I think that the loss function proposed in YOLO works quite well. Regarding classification error, I didn't do a test so I'm not sure of which one is better. But from theory, cross-entropy can get away with the problem of premature gradient saturation that MSE has to deal with.

**Comment by txdat** [comment]
i'm testing with softmax cross entropy as classification loss and check no gradient for negative anchors. but  when training with adam & learning rate decayf from 1e-4, classification loss rises to a larger value after 4k iterations, i dont know how to fix it, can you give me some solutions?

**Comment by txdat** [comment]
during training, i see bounding box regression convergence faster than iou and classify, but after 30 epochs  with voc07 (and loss not decreasing after), output of bounding boxes only default anchors trained from warmup epochs, should i train it with more epochs?

**Comment by experiencor** [comment]
@txdat The general rule is that the more weight you give to a loss (whether regression, classification or confidence), the more accurate that prediction will become. You have to make a trade-off.
